logstreams/src/main/java/io/zeebe/distributedlog/impl/DefaultDistributedLogstreamService.java

```diff
@@ -175,9 +177,32 @@ public long append(String nodeId, long commitPosition, byte[] blockBuffer) {
       // error code;
     }
 
+    // if the lastAppendIndex + 1 is less then the appendIndex, we missed some events
+    if (lastAppendIndex + 1 < appendIndex) {
+      logger.trace(
+          "We received an event out of order. Last append index {} and try to append index {}, will reject this event.",
+          lastAppendIndex,
+          appendIndex);
+      return -lastAppendIndex;
+    }
+
+    if (commitPosition <= lastPosition && lastAppendIndex <= appendIndex) {
```

Comment:
Here I was not 100% sure if this can even happen

Reply:
Actually it worked fine before on the benchmark without this. But the Distributed Recovery test failed. With this addition the benchmark fails 😅

Reply:
What we want is that `lastAppendIndex` is updated to the appendIndex of last append. When we use  `lastAppendIndex <= appendIndex`, `lastAppendIndex` will be `appendIndex + 1`. It wouldn't matter if, instead of `lastAppendIndex++`,  we just assign the value `lastAppendIndex = appendIndex`.

Reply:
So do you mean in general `lastAppendIndex = appendIndex.` or just for this case (i think I had this once)? Which solution you would like to have :smile:

Reply:
If it is possible to use everywhere `lastAppendIndex = appendIndex`, it would be better. It is more understandable for me. But, just here would also be enough :slightly_smiling_face:

Reply:
Agree I will verify that